packages/terra-date-picker/src/DateUtil.js

```diff
@@ -2,34 +2,48 @@
 import moment from 'moment';
 
 class DateUtil {
-  // Converts an ISO 8601 date into a moment object. If the date is invalid and unable to convert, the originally provided date is returned.
+  /**
+   * Creates a moment object using the provided date string.
+   * @param {string|undefined} date - The date to convert. Expect to be in ISO format.
+   * @return {object|undefined} - The moment object. Undefined if unable to convert.
+   */
   static createSafeDate(date) {
-    if (date) {
-      const momentDate = moment(date);
-      return momentDate.isValid() ? momentDate : date;
+    if (!date || (date && date.length === 0)) {
+      return undefined;
     }
 
-    return date;
+    const momentDate = moment(date);
+    return momentDate.isValid() ? momentDate : undefined;
   }
 
-  // Filters out any invalid dates in the provided list of dates and returns a list of moment objects representation of the valid dates
+  /**
+   * Filters any invalid dates in the provided list of dates
+   * @param {array} date - The array of dates to filter.
+   * @return {array|undefined} - The array of moment objects each representing the valid dates. Undefined if there are no valid dates.
+   */
   static filterInvalidDates(dates) {
-    const momentDates = [];
+    const validMomentDates = [];
 
-    if (dates) {
+    if (Array.isArray(dates)) {
       let index = 0;
       for (index = 0; index < dates.length; index += 1) {
         const momentDate = moment(dates[index]);
         if (momentDate.isValid()) {
-          momentDates.push(momentDate);
+          validMomentDates.push(momentDate);
         }
       }
     }
 
-    return momentDates.length > 0 ? momentDates : dates;
+    return validMomentDates.length > 0 ? validMomentDates : undefined;
   }
 
-  // Checks if a given date is out of the range between the start and end dates.
+  /**
+   * Determines if a date if in the range of two given dates.
+   * @param {object} sourceDate - The moment date to check if it is within range.
+   * @param {object} minDate - The moment date that represents the minimum date of the range.
+   * @param {object} maxDate - The moment date that represents the maximum date of the range.
```

Comment:
A few of these params don't match the method signature below

Reply:
In [PR #578](https://github.com/cerner/terra-framework/pull/578/files#diff-bed1c9aab9f3944cbb524e8668862592), I renamed the name of these parameters to minDate and maxDate. I'm using the updated names now so that I don't have to go back and update again.

Reply:
Since you're not making any changes to this function in this PR, I think it makes more sense to put the beefed up doc in #578

Reply:
Reverted the documentation and will add it using PR #578 
https://github.com/cerner/terra-framework/pull/607/commits/28984dc7cbf6d82a2063bc3e0b8c017f0d6e98f2